### ISSUE TITLE
Parse difference for KSH, no semicolon required after double brackets

### ISSUE THREAD

**Issue by masalomon** [koalaman/shellcheck#1099]
#### For bugs
- Rule Id [SC1049](https://github.com/koalaman/shellcheck/wiki/SC1049), and possibly others.
- My shellcheck version (`shellcheck --version` or "online"): **online**
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/ksh
if [[ -z "" ]] then echo yes ; fi
```

#### Here's what shellcheck currently says:
[SC1049](https://github.com/koalaman/shellcheck/wiki/SC1049): Did you forget the 'then' for this 'if'?
[SC1073](https://github.com/koalaman/shellcheck/wiki/SC1073): Couldn't parse this if expression. Fix to allow more checks.
[SC1010](https://github.com/koalaman/shellcheck/wiki/SC1010): Use semicolon or linefeed before 'then' (or quote to make it literal).
[SC1050](https://github.com/koalaman/shellcheck/wiki/SC1050): Expected 'then'.
[SC1072](https://github.com/koalaman/shellcheck/wiki/SC1072): Unexpected keyword/token. Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:
Nothing. This is valid in Ksh, as far back as ksh88. The semicolon is not required before the `then` if the `if` condition uses double brackets.

See related StackOverflow question: https://stackoverflow.com/q/48107383/1313898
